plugins/modules/user.py

```diff
@@ -0,0 +1,224 @@
+    username: "test_collection_user"
+    state: absent
+
+- name: Update the username
+  ansible.eda.user:
+    username: "test_collection_user"
+    new_username: "test_collection_user_updated"
+    first_name: "Test Collection User"
+    last_name: "Test Collection User"
+    email: "test@test.com"
+    password: "test"
+    roles:
+      - "Admin"
+"""
+
+
+RETURN = """
+id:
+    description: ID of the managed AWX token.
+    returned: when state is 'present' and successful
+    type: str
+    sample: "123"
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ansible.eda.plugins.module_utils.arguments import AUTH_ARGSPEC
+from ansible_collections.ansible.eda.plugins.module_utils.client import Client
+from ansible_collections.ansible.eda.plugins.module_utils.controller import Controller
+from ansible_collections.ansible.eda.plugins.module_utils.errors import EDAError
+
+
+def main():
+    argument_spec = dict(
+        username=dict(required=True),
+        new_username=dict(),
+        first_name=dict(),
+        last_name=dict(),
+        email=dict(),
+        password=dict(no_log=True),
+        roles=dict(
+            type="list",
+            elements="str",
+            choices=[
+                "Admin",
+                "Editor",
+                "Contributor",
+                "Operator",
+                "Auditor",
+                "Viewer",
+            ],
+        ),
+        update_secrets=dict(type="bool", default=True),
+        state=dict(choices=["present", "absent"], default="present"),
+    )
+
+    argument_spec.update(AUTH_ARGSPEC)
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    client = Client(
+        host=module.params.get("controller_host"),
+        username=module.params.get("controller_username"),
+        password=module.params.get("controller_password"),
+        timeout=module.params.get("request_timeout"),
+        validate_certs=module.params.get("validate_certs"),
+    )
+
+    controller = Controller(client, module)
+    user_endpoint = "users"
+
+    # Extract the params
+    username = module.params.get("username")
+    new_username = module.params.get("new_username")
+    roles = module.params.get("roles")
+    state = module.params.get("state")
+    ret = {}
+
+    try:
+        user_type = controller.get_one_or_many(user_endpoint, name=username)
+    except EDAError as eda_err:
+        module.fail_json(msg=str(eda_err))
+
+    if state == "absent":
+        # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
+        try:
+            ret = controller.delete_if_needed(user_type, endpoint=user_endpoint)
+        except EDAError as eda_err:
+            module.fail_json(msg=str(eda_err))
+        module.exit_json(**ret)
+
+    role_id = []
+    if roles:
+        for role in roles:
+            role_identifier = controller.resolve_name_to_id("roles", role)
```

Comment:
I would enclose in a try except block to catch the case when the role does not exist.